src/index.ts

```diff
@@ -1,310 +1,312 @@
-                : undefined;
-          if (!key || key[0] === '_') {
-            constant = false;
-          }
-          if (!constant) return;
-          const value = toConstant(property.value);
-          if (!constant) return;
-          result[key] = value;
-        } else if (b.isObjectMethod(property)) {
-          constant = false;
-        } else if (b.isSpreadProperty(property)) {
-          const argument = toConstant(property.argument);
-          if (!argument) constant = false;
-          if (!constant) return;
-          Object.assign(result, argument);
-        }
-      }
-      return result;
-    }
-    if (b.isParenthesizedExpression(expression)) {
-      return toConstant(expression.expression);
-    }
-    if (b.isRegExpLiteral(expression)) {
-      return new RegExp(expression.pattern, expression.flags);
-    }
-    if (b.isSequenceExpression(expression)) {
-      for (let i = 0; i < expression.expressions.length - 1 && constant; i++) {
-        toConstant(expression.expressions[i]);
-      }
-      return toConstant(
-        expression.expressions[expression.expressions.length - 1],
-      );
-    }
-    if (b.isStringLiteral(expression)) {
-      return expression.value;
-    }
-    // TODO: TaggedTemplateExpression
-    if (b.isTemplateLiteral(expression)) {
-      let result = '';
-      for (let i = 0; i < expression.quasis.length; i++) {
-        const quasi = expression.quasis[i];
-        result += quasi.value.cooked;
-        if (i < expression.expressions.length) {
-          result += '' + toConstant(expression.expressions[i]);
-        }
-      }
-      return result;
-    }
-    if (b.isUnaryExpression(expression)) {
-      const argument = toConstant(expression.argument);
-      if (!constant) {
-        return;
-      }
-      switch (expression.operator) {
-        case '-':
-          return -argument;
-        case '+':
-          return +argument;
-        case '!':
-          return !argument;
-        case '~':
-          return ~argument;
-        case 'typeof':
-          return typeof argument;
-        case 'void':
-          return void argument;
-      }
-    }
-    constant = false;
-  }
-  const result = toConstant(expression);
-  return constant ? {constant: true, result} : {constant: false};
+	expression: b.Expression,
+	options: ExpressionToConstantOptions = {},
+): { constant: true; result: any } | { constant: false; result?: void } {
+	let constant = true;
+	function toConstant(expression: b.Expression): any {
+		if (!constant) return;
+		if (b.isArrayExpression(expression)) {
+			const result = [];
+			for (let i = 0; constant && i < expression.elements.length; i++) {
+				const element = expression.elements[i];
+				if (b.isSpreadElement(element)) {
+					const spread = toConstant(element.argument);
+					if (!(isSpreadable(spread) && constant)) {
+						constant = false;
+					} else {
+						result.push(...spread);
+					}
+				} else if (b.isExpression(element)) {
```

Comment:
this check was added because  `element` was typed with null, which `toConstant` refused to accept. behavior might be altered slightly for the worse. could also just type `toConstant` as accepting nullables.